lib/color.cpp

```diff
@@ -20,27 +20,36 @@
 
 #ifndef _WIN32
 #include <unistd.h>
-#endif
 #include <cstddef>
 #include <sstream> // IWYU pragma: keep
+#endif
 
-#ifdef _WIN32
-std::ostream& operator<<(std::ostream& os, const Color& /*c*/)
+bool isStdOutATty()
 {
+#ifndef _WIN32
+    static const bool stdout_tty = isatty(STDOUT_FILENO);
```

Comment:
This is not your fault.. but I bring it up anyway. 

The idea is that we will only print color codes to a terminal right?

When I pipe results to a file with "2>" I get color codes. Should we have a `isStdErrATty()` also?

Reply:
Ah - didn't think about that. Good point. I actually ran into this before and was a bit baffled by it.

I would prefer to do that change in a different PR and add tests for it as well as release notes about the behavior change.

Reply:
If you do it in a separate PR that is excellent!